lib/Differentiator/BaseForwardModeVisitor.cpp

```diff
@@ -668,18 +668,53 @@ StmtDiff BaseForwardModeVisitor::VisitForStmt(const ForStmt* FS) {
   const Stmt* init = FS->getInit();
   StmtDiff initDiff = init ? Visit(init) : StmtDiff{};
   addToCurrentBlock(initDiff.getStmt_dx());
-  VarDecl* condVarDecl = FS->getConditionVariable();
-  VarDecl* condVarClone = nullptr;
-  if (condVarDecl) {
-    DeclDiff<VarDecl> condVarResult = DifferentiateVarDecl(condVarDecl);
-    condVarClone = condVarResult.getDecl();
-    if (condVarResult.getDecl_dx())
-      addToCurrentBlock(BuildDeclStmt(condVarResult.getDecl_dx()));
+
+  StmtDiff condDiff = Clone(FS->getCond());
+  Expr* cond = condDiff.getExpr();
+
+  // Declaration in the condition (if any) needs to be differentiated.
+  if (VarDecl* condVarDecl = FS->getConditionVariable()) {
+    VarDecl* condVarClone =
+        BuildVarDecl(condVarDecl->getType(), condVarDecl->getNameAsString(),
+                     Clone(condVarDecl->getInit()), condVarDecl->isDirectInit(),
+                     nullptr, condVarDecl->getInitStyle());
+    VarDecl* _d_condVarDecl = BuildVarDecl(
+        condVarClone->getType(), "_d_" + condVarClone->getNameAsString(),
+        nullptr, condVarClone->isDirectInit(), nullptr,
```

Comment:
```suggestion
        /*Init=*/nullptr, condVarClone->isDirectInit(), nullptr,
```
We comment in the the arguments which are unclear what parameters refer to.